README.md
# websitebaker_sitemap

Allows you to easily create a sitemap. You can even work with additional layout settings for the sitemap. 

 Website Baker Project <http://www.websitebaker.org/>
 Copyright (C) 2004-2010, Ryan Djurovich

 Website Baker is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2 of the License, or
 (at your option) any later version.

 Website Baker is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with Website Baker; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 
 -----------------------------------------------------------------------------------------
  Sitemap module for Website Baker v2.7.x-2.8.x (http://www.websitebaker.org)
  The Sitemap module provides the facility to add a sitemap of your website.
 -----------------------------------------------------------------------------------------

[Discussion forum](http://www.websitebaker2.org/forum/index.php/topic,1798.msg126681.html#msg126681)

Also see http://www.websitebakers.com/pages/modules/listings/section-pages/sitemap.php


CHANGELOG.md
   DEVELOPMENT HISTORY (Change Log):

   v3.1.3 	2011-08-09	Dietrich Roland Pehlke
						- remove old WB 2.4 stuff, vars inside the view.php

   v3.1.2 (Michael Milette; August 9, 2010)
                   Fixed: (with error reporting on 'E_ALL'), notice message
                     regarding undefined index: show_hidden that appeared when
                     saving settings.
                   Updated this changelog using info from help.php.
                   Remove change log from help.php to minimize techy info in
                     user interface.

   v3.1.1 (Michael Milette; August 8, 2010) No new features.
                   [update_sitemap.php] Now replaced by new upgrade.php which is automatically
                     run when upgrading using Modules installation tool.
                   [upgrade.php] Upgrades now include the creation of the show_hidden field.
                   Upgrades now reports errors.
                   [info.php]    Now reflects compatibility with Website Baker v2.8.x.
                   Updated: All copyright notices upgraded to now include 2010.

   v3.1            (kozmoz; 10-16-2006)
                   Added: Option to show hidden pages.
                   Added: Dutch Translation

   v3.0   (rssmith, Ruebenwurzel; 9-12-2006)
                   Fixed: [MODIFIED_BY] displays now the displayname instead of
                     the userid.
                   Fixed: install script now works with mysql5 strict mode

   v2.9   (Ruebenwurzel; 05-07-2006)
                   Added: New variables: MODIFIED_DATE] and [MODIFIED_TIME]
                     setup to allow for a more controlled layout
                   Changed: The date/time is now more customizable.

   v2.8   (Ruebenwurzel; 03-04-2006)
                   Renamed: databasefield 'loop' to 'sitemaploop' to be
                     compatible to MySQL 5.x
                   Added: update_sitemap.php script to update from previous
                     versions
                   Updated: All copyright notices includes now 2006
                   Added: Default value to all varchar and text fields

   v2.7   (Ruebenwurzel; 12-28-2005)
                   Added: New multilanguage support
                   Changed: Minor layout changes in admin interface

   v2.6.2 (Ruebenwurzel; 11-25-2005)
                   Added: Language file works with help file
                   Changed: Minor layout changes in admin Interface
                   Fixed: Code cleaning and all files stored UNIX conform

   v2.6.1 (Woudloper,Ruebenwurzel; 11-24-2005)
                   Added: Uninstall functionality for the module.
                   Fixed: Layout of the help.php file within Internet Explorer.

   v2.6.0 (Woudloper; 11-16-2005)
                   Added: External language file</li>
                   Fixed: Declaration of variables. On some servers (with error
                     reporting on 'E_ALL') warnings where displayed about some
                     variables.
                   Fixed: Warning message (regarding variables) when saving the
                     settings within the admin console.
                   Changed: Layout of the admin section for the sitemap module.



help.php
<?php
/*
 Website Baker Project <http://www.websitebaker.org/>
 Copyright (C) 2004-2010, Ryan Djurovich

 Website Baker is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2 of the License, or
 (at your option) any later version.

 Website Baker is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with Website Baker; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

// STEP 1:	Initialize
require('../../config.php');

// Include WB admin wrapper script
require(WB_PATH.'/modules/admin.php');

// Load Language file
if(LANGUAGE_LOADED) {
	if(!file_exists(WB_PATH.'/modules/sitemap/languages/'.LANGUAGE.'.php')) {
		require_once(WB_PATH.'/modules/sitemap/languages/EN.php');
	} else {
		require_once(WB_PATH.'/modules/sitemap/languages/'.LANGUAGE.'.php');
	}
}

// STEP 2:	Get actual settings from database
$query_settings = $database->query("SELECT * FROM ".TABLE_PREFIX."mod_sitemap WHERE section_id = '$section_id'");
$settings = $query_settings->fetchRow();

// STEP 3:	Display the help page.
?>

<style type="text/css">
.setting_name {
	vertical-align: top;
}

.row_a a {
	 border-bottom: 1px dashed #000;
}

.row_a a:hover {
	 border-bottom: 1px solid #000;
}
</style>
<table class="row_a" cellpadding="2" cellspacing="0" border="0" align="center" width="100%">
	<tr>
		<td class="setting_name">
			<h3>Help about Sitemap Module</h3>
			
			<p>This file contains help about the sitemap module and how the different fields within the module can be used.</p>
			
			<p>The help is divided in 3 sections: <a href="#main_settings"><?php echo $SMTEXT['MSETTINGS']; ?></a>, <a href="#layout_settings"><?php echo $SMTEXT['LSETTINGS']; ?></a> and the <a href="#changelog">changelog</a>. All this information is described down below.</p>
			
			<a name="main_settings"></a><h4><?php echo $SMTEXT['MSETTINGS']; ?></h4>
			<p>This section contains the required fields that are neccasery to get this module to work.</p>
			<ul>
				<li><code><?php echo $SMTEXT['DISPLAY']; ?></code> With this option you can specify of which part of the website you would like the create a sitemap, the options ares:
					<ul>
						<li><code><?php echo $SMTEXT['WHOLESITE']; ?></code> - This is the default option. When selecting this option all the visible pages will be rendered into the sitemap.</li>
						<li><code><?php echo $SMTEXT['CHILDREN_CP']; ?></code> - As mentioned in the option, this option will only show children (if any) of the current page.</li>
						<li><code><?php echo $SMTEXT['PEERS_CP']; ?></code> - This option means that all the peers (pages at same level and below) will be displayed based upon the current page.</li>
					</ul>
				</li>
				<li><code><?php echo $SMTEXT['PAGE_LEVEL_LIMIT']; ?></code> This fields specifies the maximum depth of the sitemap that should be shown.</li>
			</ul>
				
			<a name="layout_settings"></a><h4><?php echo $SMTEXT['LSETTINGS']; ?></h4>
			<p>With the options down below you can modify the layout of the different sections for the sitemap module.</p>
			<ul>
				<li><code><?php echo $SMTEXT['HEADER']; ?></code> This field is used to render it at the beginning of a list. You can use this e.g. for working with a <code>&lt;div&gt;</code> or with specific stylesheets definitions.</li>
				<li><code><?php echo $SMTEXT['LHEADER']; ?></code> This field is used to render it at the beginning of a sitemap level. You can use this e.g. for working with a <code>&lt;div&gt;</code> or with specific stylesheets definitions. </li>
				<li><code><?php echo $SMTEXT['LLOOP']; ?></code> This template is used to display the different levels that are available for the sitemap. In order to modify this layout you can use the following tags:
					<ul>
						<li><code>[PAGE_ID]</code> - The page id of the specific page in the sitemap.</li>
						<li><code>[PARENT]</code>  - The parent of the page that is currently in the loop.</li>
						<li><code>[LINK]</code> - The link to the specific page within the sitemap.</li>
						<li><code>[PAGE_TITLE]</code> - The title of the page within the sitemap.</li>
						<li><code>[MENU_TITLE]</code> - The title of the page as it is shown witin the menu.</li>
						<li><code>[DESCRIPTION]</code> - The description of the page within the sitemap.</li>
						<li><code>[KEYWORDS]</code> - The keywords that are given for the current page within the sitemap.</li>
						<li><code>[TARGET]</code> - The target to which the page should be opened.</li>
						<li><code>[MODIFIED_WHEN]</code> - The date and time when the page has been last modified.</li>
						<li><code>[MODIFIED_DATE]</code> - The date when the page has been last modified.</li>
						<li><code>[MODIFIED_TIME]</code> - The time when the page has been last modified.</li>
						<li><code>[MODIFIED_BY]</code> - The person who made the last modification to the specific page.</li>
					</ul>
				</li>
				<li><code><?php echo $SMTEXT['LFOOTER']; ?></code> This is used to display the footer for a level. Mostly this is only used for e.g. closing a <code>&lt;/div&gt;</code>.</li>
				<li><code><?php echo $SMTEXT['FOOTER']; ?></code> This field is used to render it at the end of the sitemap. You can use this e.g. for working with a <code>&lt;div&gt;</code> tag.</li>
			</ul>
			
			<a name="changelog"></a><h4>Changelog</h4>
      <p>Change log has been moved to info.php.</p>
		</td>
	</tr>
</table>
<table cellpadding="0" cellspacing="0" border="0" width="100%">	
	<tr>
		<td>
			<input type="button" value="<?php echo $TEXT['BACK']; ?>" onclick="javascript: window.location = '<?php echo ADMIN_URL; ?>/pages/modify.php?page_id=<?php echo $page_id; ?>';" style="width: 100px; margin-top: 5px;" />
		</td>
	</tr>	
</table>

index.php
<?php
/*
 Website Baker Project <http://www.websitebaker.org/>
 Copyright (C) 2004-2010, Ryan Djurovich

 Website Baker is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2 of the License, or
 (at your option) any later version.

 Website Baker is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with Website Baker; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

header('Location: ../index.php');

?>

delete.php
<?php
/*
 Website Baker Project <http://www.websitebaker.org/>
 Copyright (C) 2004-2010, Ryan Djurovich

 Website Baker is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2 of the License, or
 (at your option) any later version.

 Website Baker is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with Website Baker; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

// Delete page from mod_sitemap
$database->query("DELETE FROM ".TABLE_PREFIX."mod_sitemap WHERE section_id = '$section_id'");

?>

upgrade.php
<?php
/*
 Website Baker Project <http://www.websitebaker.org/>
 Copyright (C) 2004-2010, Ryan Djurovich

 Website Baker is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2 of the License, or
 (at your option) any later version.

 Website Baker is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with Website Baker; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

$dbtable = TABLE_PREFIX.'mod_sitemap';

//Prevent this file from being accessed directly
if (defined('WB_PATH') == false) {
	exit("Cannot access this file directly");
}

require_once(WB_PATH.'/framework/functions.php');
$database = new database();

// Setup styles to help id errors
echo'
<style type="text/css">
.good {color: green;}
.bad  {color: red;}
.ok   {color: blue;}
.warn {color: yellow;}
</style>
';

// Version to be installed is the same or older than currently installed version
if ($module_version >= $new_module_version) {
	echo '<span class="bad">';
	$admin->print_error($MESSAGE['GENERIC']['ALREADY_INSTALLED']);
	echo '</span><br />';
	return;
}
// UPGRADE TO VERSION 3.1.1
// ************************

$result = 0;
if ($module_version < $new_module_version) {
	// Title: Upgrading to
	echo'<h3>SITEMAP - Upgrading to version '.$new_module_version.':</h3>';

	// Get ITEMS table to see what needs to be created or modified
	$itemstable = $database->query("SELECT * FROM `".$dbtable."`");
	$items = $itemstable->fetchRow();

  echo "<BR><B>Trying to rename database field '<i>loop</i>' to '<i>sitemaploop</i>'...</B><BR>";

  if (!array_key_exists('sitemaploop', $items)){
  		if ($database->query("ALTER TABLE `".$dbtable."` CHANGE `loop` `sitemaploop` TEXT NOT NULL")) {
  			echo '<span class="good">Database field <i>loop</i> renamed to <i>sitemaploop</i> successfully</span><br />';
  		} else { echo '<span class="bad">'.mysql_error().'</span><br />'; $result = 1;}
  } else { echo '<span class="ok">Database field <i>sitemaploop</i> already exists, update not needed.</span><br />';}

  echo "<BR><B>Trying to add database field '<i>show_hidden</i>'...</B><BR>";

  if (!array_key_exists('show_hidden', $items)){
  		if ($database->query("ALTER TABLE `".$dbtable."` ADD `show_hidden` INT NOT NULL")) {
  			echo '<span class="good">Database field <i>show_hidden</i> added successfully</span><br />';
  		} else { echo '<span class="bad">'.mysql_error().'</span><br />'; $result = 1;}
  } else { echo '<span class="ok">Database field <i>show_hidden</i> already exists, update not needed.</span><br />'; }

  if($result == 1) die();
}
?>

uninstall.php
<?php
/*
 Website Baker Project <http://www.websitebaker.org/>
 Copyright (C) 2004-2010, Ryan Djurovich

 Website Baker is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2 of the License, or
 (at your option) any later version.

 Website Baker is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with Website Baker; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

if(!defined('WB_PATH')) { exit("Cannot access this file directly"); }

$database->query("DROP TABLE ".TABLE_PREFIX."mod_sitemap");

?>

info.php
<?php
/*
 Website Baker Project <http://www.websitebaker.org/>
 Copyright (C) 2004-2010, Ryan Djurovich

 Website Baker is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2 of the License, or
 (at your option) any later version.

 Website Baker is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with Website Baker; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

/*
 -----------------------------------------------------------------------------------------
  Sitemap module for Website Baker v2.7.x-2.8.x (http://www.websitebaker.org)
  The Sitemap module provides the facility to add a sitemap of your website.
 -----------------------------------------------------------------------------------------

	DEVELOPMENT HISTORY (Change Log):
	
- 3.1.3 	2011-08-09	Dietrich Roland Pehlke
						- remove old WB 2.4 stuff, vars inside the view.php

   v3.1.2 (Michael Milette; August 9, 2010)
                   Fixed: (with error reporting on 'E_ALL'), notice message
                     regarding undefined index: show_hidden that appeared when
                     saving settings.
                   Updated this changelog using info from help.php.
                   Remove change log from help.php to minimize techy info in
                     user interface.

   v3.1.1 (Michael Milette; August 8, 2010) No new features.
   + [update_sitemap.php] Now replaced by new upgrade.php which is automatically
                     run when upgrading using Modules installation tool.
	 + [upgrade.php] Upgrades now include the creation of the show_hidden field.
                   Upgrades now reports errors.
   + [info.php]    Now reflects compatibility with Website Baker v2.8.x.
                   Updated: All copyright notices upgraded to now include 2010.

   v3.1            (kozmoz; 10-16-2006)
                   Added: Option to show hidden pages.
                   Added: Dutch Translation

   v3.0   (rssmith, Ruebenwurzel; 9-12-2006)
                   Fixed: [MODIFIED_BY] displays now the displayname instead of
                     the userid.
                   Fixed: install script now works with mysql5 strict mode

   v2.9   (Ruebenwurzel; 05-07-2006)
                   Added: New variables: MODIFIED_DATE] and [MODIFIED_TIME]
                     setup to allow for a more controlled layout
                   Changed: The date/time is now more customizable.

   v2.8   (Ruebenwurzel; 03-04-2006)
                   Renamed: databasefield 'loop' to 'sitemaploop' to be
                     compatible to MySQL 5.x
                   Added: update_sitemap.php script to update from previous
                     versions
                   Updated: All copyright notices includes now 2006
                   Added: Default value to all varchar and text fields

   v2.7   (Ruebenwurzel; 12-28-2005)
                   Added: New multilanguage support
                   Changed: Minor layout changes in admin interface

   v2.6.2 (Ruebenwurzel; 11-25-2005)
                   Added: Language file works with help file
                   Changed: Minor layout changes in admin Interface
                   Fixed: Code cleaning and all files stored UNIX conform

   v2.6.1 (Woudloper,Ruebenwurzel; 11-24-2005)
                   Added: Uninstall functionality for the module.
                   Fixed: Layout of the help.php file within Internet Explorer.

	 v2.6.0 (Woudloper; 11-16-2005)
                   Added: External language file</li>
                   Fixed: Declaration of variables. On some servers (with error
                     reporting on 'E_ALL') warnings where displayed about some
                     variables.
                   Fixed: Warning message (regarding variables) when saving the
                     settings within the admin console.
                   Changed: Layout of the admin section for the sitemap module.

*/

$module_directory = 'sitemap';
$module_name = 'Sitemap';
$module_function = 'page';
$module_version = '3.1.3';
$module_platform = '2.5.x - 2.8.x';
$module_author = 'Ryan Djurovich, Frank Schoep, Woudloper, Ruebenwurzel, Rob Smith, Mouring Kolhoff, Michael Milette, Dietrich Roland Pehlke (last)';
$module_license = 'GNU General Public License';
$module_description = 'This module allows you to easily create a sitemap. You can even work with additional layout settings for the sitemap. It also gives you the option to show hidden menu items if required';
$module_home = 'http://www.websitebakers.com';
$module_guid = '8B7C09E5-6246-4264-8959-8AA542D7BD6';

?>